content/src/service.ts

```diff
@@ -1,20 +1,86 @@
-import { Lifecycle } from '@well-known-components/interfaces'
+import { IHttpServerComponent, Lifecycle } from '@well-known-components/interfaces'
+import { setupRouter } from './controller/routes'
 import { EnvironmentConfig } from './Environment'
 import { startSynchronization } from './logic/synchronization'
 import { migrateContentFolderStructure } from './migrations/ContentFolderMigrationManager'
-import { AppComponents } from './types'
+import { AppComponents, GlobalContext, UPLOADS_DIRECTORY } from './types'
+import path from 'path'
+import fs from 'fs'
+
+async function purgeUploadsDirectory({ logs, fs }: Pick<AppComponents, 'logs' | 'fs'>): Promise<void> {
+  const logger = logs.getLogger('purge-uploads-directory')
+  logger.info("Cleaning up the Server's uploads directory...")
+  try {
+    const directory = UPLOADS_DIRECTORY
+    const files = await fs.readdir(directory)
+    files.forEach(async (file) => {
+      await fs.unlink(path.join(directory, file))
+    })
+    logger.info('Cleaned up!')
+  } catch (e) {
+    logger.error('There was an error while cleaning up the upload directory: ', e)
+  }
+}
+
+async function setupApiCoverage(server: IHttpServerComponent<GlobalContext>) {
+  // Write object to disk because Jest runs tests in isolated environments
+  const coverageDir = path.join(__dirname, '../api-coverage')
+  try {
+    await fs.promises.access(coverageDir)
+  } catch (err) {
+    await fs.promises.mkdir(coverageDir)
+  }
+  const coverageFilePath = path.join(coverageDir, `api-coverage.csv`)
+  server.use(async (context, next) => {
+    const response = await next()
+    await fs.promises.appendFile(
+      coverageFilePath,
+      `${context.url.pathname},${context.request.method},${response.status}\n`
+    )
+    return response
+  })
+}
+
+// TODO
+// if (env.getConfig(EnvironmentConfig.VALIDATE_API) || process.env.CI === 'true') {
+//   this.app.use(
+//     OpenApiValidator.middleware({
+//       apiSpec: CONTENT_API,
+//       validateResponses: true,
+//       validateRequests: false,
+//       ignoreUndocumented: true,
+//       ignorePaths: /\/entities/
+//     })
+//   )
+// }
 
 // this function wires the business logic (adapters & controllers) with the components (ports)
 export async function main(program: Lifecycle.EntryPointParameters<AppComponents>): Promise<void> {
   const { components, startComponents } = program
+  const globalContext: GlobalContext = {
+    components
+  }
 
+  const { logs, fs } = components
   await migrateContentFolderStructure(components)
 
   // first of all, run the migrations
   await components.migrationManager.run()
 
-  // TODO: move this purgeUploadsDirectory method to a standalone function inside src/logic/ folder
-  await components.server.purgeUploadsDirectory()
+  const router = await setupRouter(globalContext)
+
+  if (process.env.API_COVERAGE === 'true') {
```

Comment:
I could have another `main` for testing purposes, but not quite sure if that's better. This shouldn't affect production performance, since it's just an if on startup.

Reply:
Yeah, just said it because I usually don't like to mess prod code with testing code and it felt weird the first time I read it. But anyways, can live with that.